tests/unit/igraph_st_mincut.c

```diff
@@ -0,0 +1,57 @@
+/*
+   IGraph library.
+   Copyright (C) 2021  The igraph development team <igraph@igraph.org>
+
+   This program is free software; you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation; either version 2 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#include <igraph.h>
+#include <assert.h>
+
+int main() {
+
+    igraph_t g;
+    igraph_vector_t cut, partition, partition2;
+    igraph_real_t value;
+    int source = 0;
+    int target = 4;
+
+    igraph_vector_init(&partition, 0);
+    igraph_vector_init(&partition2, 0);
+    igraph_vector_init(&cut, 0);
+
+    igraph_small(&g, 5, IGRAPH_DIRECTED, 0, 1, 1, 2, 1, 3, 2, 4, 3, 4, -1);
+
+    igraph_st_mincut(&g, &value, &cut, &partition, &partition2, source, target, /*capacity*/ 0);
+
+    /*     tests     */
+
+    assert(igraph_vector_size(&cut) == 1);
+    assert(igraph_vector_size(&cut) == 1);
+    assert(igraph_vector_size(&partition2) == 4);
+
+    int cut_edge = VECTOR(cut)[0];
+    int partition_vertex = VECTOR(partition)[0];
```

Comment:
Please move _declarations_ to the beginning of the block, to comply with the C90 standard.

We will probably start allowing mid-block declarations soon, but the decision has not been made yet.

@ntamas Any opinions on this?

Reply:
I feel a general uneasiness about mid-block declarations even if C90 allows it. If we always declare everything at the top of a block, then it is easy to get an overview of the variables that a particular block uses as one only needs to look them up at the top of the block. It becomes harder if the variables can be declared anywhere (although syntax highlighting obviously helps).

Reply:
C90 (same as C89) does not allow it. Older (but not very old) MSVC doesn't allow it. Only C99 and new MSVC allows it.

OK, we keep things at the top of the block then.

When moving declarations mid-block would improve readability, there is an easy workaround: add a new block, i.e.

```
{
     int i;

    {
        int a=1;
        f(a);
    }

    ...

    {
        int b=3;
        f(b);
     }
}
```

An added readability advantage is that these extra syntactic blocks can clearly delimit semantic blocks in the code.